src/echo.ts

```diff
@@ -46,6 +46,19 @@ async function echoServer(client: TCPSocket) {
   }
 
   let byteCount = 0;
+
+  if (!readable) {
```

Comment:
This is a bug in the TypeScript definitions. `readable` and `writable` will always be present. Please patch the definitions before landing this change.

Reply:
I agree that the spec should be updated. It's easy for the WebIDL definitions for dictionaries returned by methods (rather than consumed by methods) to get out of sync with reality because the actual source of truth for how they are initialized comes from the method steps.